migrate_nidirect_utils/src/MigrationProcessors.php

```diff
@@ -381,4 +390,69 @@ public function flags($entity_type) {
     return $output;
   }
 
+  /**
+   * Import audit data.
+   *
+   * @param string $entity_type
+   *   The entity type to process.
+   *
+   * @return string
+   *   Information/results of on the process.
+   */
+  public function audit($entity_type) {
+
+    // Only process the entity types listed in the array.
+    if (!in_array($entity_type, ['article', 'contact', 'page'])) {
+      return "Audit processing for " . $entity_type . " is not enabled.";
+    }
+
+    $d7_audit_nids = $this->dbConnMigrate->query("
+      SELECT f.entity_id 
+      FROM flagging f
+      JOIN node n
+      ON f.entity_id = n.nid
+      WHERE n.type = '$entity_type'
+      AND f.fid = 1
+    ")->fetchCol(0);
+
+    $today = date('Y-m-d', \Drupal::time()->getCurrentTime());
+
+    // Fetch current or future audit dates.
+    $excluded_audit_nids = $this->dbConnDrupal8->query("
+      SELECT a.entity_id
+      FROM node__field_next_audit_due AS a
+      JOIN node n
+      ON a.entity_id = n.nid
+      WHERE STR_TO_DATE(field_next_audit_due_value, '%Y-%m-%d') >= DATE_FORMAT(NOW(), '%Y-%m-%d')
```

Comment:
As discussed, think you should remove this date check as it isn't necessary

Reply:
Conversation needed regarding the migration of D7 data and updating of existing migrated content.